### PR TITLE
[TC] Create Environment registry extension for CapsulaHub

### PR DESCRIPTION
**Pull request by MikeSquall**
https://github.com/capsulajs/capsulahub-extensions/issues/25

### CODE REVIEW

File: packages/environment-registry/tests/features/bootstrap.feature [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+Scenario: EnvRegistry extension tries to register with bad configuration
+  Given  a Workspace with EnvRegistry extension
+  When  EnvRegistry extension tries to register with a configuration with invalid format
+  Then  an error is thrown
+  
+Scenario: EnvRegistry registers the provided environment
+  Given A Workspace with EnvRegistry extension
+  And  EnvRegistry with register method and a given environment
+  When EnvRegistry extension registers itself to the Workspace
```
Comment (MikeSquall) [inline packages/environment-registry/tests/features/bootstrap.feature]:
```suggestion
  And EnvRegistry extension registers itself to the Workspace
```

File: packages/environment-registry/tests/features/bootstrap.feature [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  Then  an EnvRegistry instance is available
+
+Scenario: EnvRegistry extension tries to register with bad configuration
+  Given  a Workspace with EnvRegistry extension
+  When  EnvRegistry extension tries to register with a configuration with invalid format
+  Then  an error is thrown
+  
+Scenario: EnvRegistry registers the provided environment
+  Given A Workspace with EnvRegistry extension
+  And  EnvRegistry with register method and a given environment
```
Comment (MikeSquall) [inline packages/environment-registry/tests/features/bootstrap.feature]:
```suggestion
  And  An environment with `envKey: dev` and `env: 'myEnv'`
```

File: packages/environment-registry/tests/features/bootstrap.feature [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  When EnvRegistry extension registers itself to the Workspace
+  And  EnvRegistry registers the environment
```
Comment (MikeSquall) [inline packages/environment-registry/tests/features/bootstrap.feature]:
```suggestion
  When  EnvRegistry registers the environment
```

File: packages/environment-registry/tests/features/bootstrap.feature [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  And  an EnvRegistry instance with this environment is available
```
Comment (MikeSquall) [inline packages/environment-registry/tests/features/bootstrap.feature]:
```suggestion
  Then  Registration of the environment is performed with success
  And   Subscribing to environments method returns 'myEnv'
```

File: packages/environment-registry/tests/features/bootstrap.feature [hunk 5]
```diff
@@ -0,0 +1,19 @@
+Feature: Create Environment registry extension for CapsulaHub
+
+Scenario: EnvRegistry extension creates new instance of EnvRegistry while registering itself
+  Given A Workspace with EnvRegistry extension
+  When  EnvRegistry extension registers itself to the Workspace
+  Then  An EnvRegistry instance is available
+
+Scenario: EnvRegistry extension tries to register with bad configuration
```
Comment (dmitriyslaym) [inline packages/environment-registry/tests/features/bootstrap.feature]: don't need this case, it's us who make the request ion the bootstrap of extension, so it's our duty to make it correctly once.

File: packages/environment-registry/tests/features/bootstrap.feature [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  Given A Workspace with EnvRegistry extension
+  When  EnvRegistry extension tries to register with a configuration with invalid format
+  Then  An error is thrown
+  
+Scenario: EnvRegistry registers the provided environment
```
Comment (dmitriyslaym) [inline packages/environment-registry/tests/features/bootstrap.feature]: I'm not sure that we need it here - we've already checked that we receive an instance of EnvRegistry service. So we expect to all of its functionality work correctly - it has been tested in this service implementation itself.

**Review by dmitriyslaym — CHANGES_REQUESTED** [review]
Suggest to add a scenario that this bootstrap function that is imported from the extension is resolved before the workspace.registerService resolves (that it doesn't wait workspace.registerService to be resolved internally).

**Review by dmitriyslaym — CHANGES_REQUESTED** [review]
Scenario: EnvRegistry extension bootstrap function resolves correctly and triggers the registration of an instance of EnvRegistry in Workspace
  Given EnvRegistry extension
  And   A valid EnvRegistry configuration with a token and configProvider
  And   Workspace
  When  EnvRegistry extension bootstrap function is called with a Workspace as a first argument and envRegistryService configuration as a second argument
  Then  Correct EnvRegistry configuration is applied while creating an instance of EnvRegistryService
  And   The registration of EnvRegistryService in Workspace is triggered with the correct arguments
  And   The promise, that is returned from the call of the bootstrap function, resolves with no args
  And   The returned promise from a call of registration function is resolved (only after the resolve of bootstrap function)

Scenario: EnvRegistry extension bootstrap function rejects with an error if the creation of an instance of EnvRegistry throws an error
  Given EnvRegistry extension
  And   An invalid EnvRegistry configuration with a token and configProvider
  And   Workspace
  When  EnvRegistry extension bootstrap function is called with a Workspace as a first argument and envRegistryService configuration as a second argument
  And   The creation of EnvRegistryService instance fails with an error "Error1"
  Then  The promise, that is returned from the call of the bootstrap function, rejects with an error "Error1"
  And   The registration of the service in Workspace was not triggered

I've explained all the desired tests in Gherkin syntax, please make some arrangements to the syntax is it's required - and it will be the correct scenarios.

**Review by dmitriyslaym — APPROVED** [review]

**Review by IrinaComindaru — APPROVED** [review]

**Pull request merged**